src/index.ts

```diff
@@ -17,11 +17,27 @@ export const angularModule = angular.module(name, [])
 
         init: function($scope, el): void {
           if (!el.hasAttribute('popgun-listening')) {
-            el.addEventListener('PopgunContentSetup', function(e) {
-              let pop = popgun.getPopFromGroupId((<Element>e.target).getAttribute('popgun-group'));
-              $compile(pop.popOver.element)($scope);
-              $scope.$apply();
-            }, false);
+            el.addEventListener('PopgunContentSetup', (function($compileScope) {
+              return function (e) {
+                  let $newScope = $compileScope.$new();
```

Comment:
could there be cases where the ui tries to set a property on the scope thinking it will affect the parent but in reality it ends up shadowing the parent? i suppose that's always a possibility with scope inheritance, but it's worth considering. that's why I was saying we'd have to assume the scope passed in was destroyable rather than creating a new one when we chatted @matthewHuiCsc